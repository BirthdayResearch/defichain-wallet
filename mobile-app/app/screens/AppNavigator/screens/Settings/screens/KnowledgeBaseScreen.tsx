import {
  ThemedIcon,
  ThemedScrollView,
  ThemedSectionTitle,
  ThemedText,
  ThemedTouchableOpacity
} from '@components/themed'
import { StackScreenProps } from '@react-navigation/stack'
import { tailwind } from '@tailwind'
import { translate } from '@translations'
import { SettingsParamList } from '../SettingsNavigator'
import { useFeatureFlagContext } from '@contexts/FeatureFlagContext'

type Props = StackScreenProps<SettingsParamList, 'KnowledgeBaseScreen'>

export function KnowledgeBaseScreen ({ navigation }: Props): JSX.Element {
  const { isFeatureAvailable } = useFeatureFlagContext()
  const knowledgeBaseItems = [
    {
      label: 'Recovery words',
      testID: 'recovery_words_faq',
      onPress: () => navigation.navigate('RecoveryWordsFaq')
    }, {
      label: 'Passcode',
      testID: 'passcode_faq',
      onPress: () => navigation.navigate('PasscodeFaq')
    }, {
      label: 'UTXO vs Token',
      testID: 'utxo_and_token_faq',
      onPress: () => navigation.navigate('TokensVsUtxo')
    },
    {
      label: 'DEX',
      testID: 'dex_faq',
      onPress: () => navigation.navigate('DexFaq')
    },
    {
      label: 'Liquidity Mining',
      testID: 'liquidity_mining_faq',
      onPress: () => navigation.navigate('LiquidityMiningFaq')
    },
    {
      label: 'Loans',
      testID: 'loans_faq',
      onPress: () => navigation.navigate({
        name: 'LoansFaq',
        params: {
          activeSessions: [0]
        }
      })
    }
  ]

  if (isFeatureAvailable('auction')) {
    knowledgeBaseItems.push({
      label: 'Auctions',
      testID: 'auctions_faq',
      onPress: () => navigation.navigate('AuctionsFaq')
    })
  }

  return (
    <ThemedScrollView
      style={tailwind('flex-1 pb-8')}
      testID='knowledge_base_screen'
    >
      <ThemedSectionTitle
        testID='knowledge_base_title'
        text={translate('screens/KnowledgeBase', 'LEARN ABOUT')}
      />

      {knowledgeBaseItems.map((each) =>
        <NavigateItemRow key={each.label} {...each} />
      )}
    </ThemedScrollView>
  )
}

function NavigateItemRow ({
  testID,
  label,
  onPress
}: { testID: string, label: string, onPress: () => void }): JSX.Element {
  return (
    <ThemedTouchableOpacity
      onPress={onPress}
      style={tailwind('flex p-4 pr-2 flex-row items-center justify-between')}
      testID={testID}
    >
      <ThemedText style={tailwind('font-medium')}>
        {translate('screens/KnowledgeBase', label)}
      </ThemedText>

      <ThemedIcon
        iconType='MaterialIcons'
        name='chevron-right'
        size={24}
      />
    </ThemedTouchableOpacity>
  )
}
