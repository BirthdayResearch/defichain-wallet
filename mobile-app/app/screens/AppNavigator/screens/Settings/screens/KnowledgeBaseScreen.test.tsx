import { render } from '@testing-library/react-native'
import { KnowledgeBaseScreen } from './KnowledgeBaseScreen'

jest.mock('@shared-contexts/ThemeProvider')
jest.mock('@contexts/FeatureFlagContext')

describe('knowledge base screen', () => {
  it('should render', async () => {
    const navigation: any = {
      navigate: jest.fn()
    }
    const route: any = {}
    const rendered = render(<KnowledgeBaseScreen navigation={navigation} route={route} />)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
