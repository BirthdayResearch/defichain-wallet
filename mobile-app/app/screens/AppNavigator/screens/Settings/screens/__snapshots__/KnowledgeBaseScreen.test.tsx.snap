// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`knowledge base screen should render 1`] = `
<RCTScrollView
  style={
    Array [
      Object {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 32,
      },
      Object {
        "backgroundColor": "rgba(245, 245, 245, 1)",
      },
    ]
  }
  testID="knowledge_base_screen"
>
  <View>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "color": "rgba(115, 115, 115, 1)",
              "fontFamily": "MediumFont",
              "fontSize": 12,
              "fontWeight": "500",
              "lineHeight": 16,
              "paddingBottom": 8,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 24,
            },
            Object {
              "backgroundColor": "rgba(245, 245, 245, 1)",
              "color": "rgba(115, 115, 115, 1)",
            },
          ],
        ]
      }
      testID="knowledge_base_title"
      text="LEARN ABOUT"
    >
      LEARN ABOUT
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderBottomColor": "rgba(229, 229, 229, 1)",
          "borderBottomWidth": 1,
          "borderLeftColor": "rgba(229, 229, 229, 1)",
          "borderRightColor": "rgba(229, 229, 229, 1)",
          "borderTopColor": "rgba(229, 229, 229, 1)",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 8,
          "paddingTop": 16,
        }
      }
      testID="recovery_words_faq"
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "fontFamily": "MediumFont",
                "fontWeight": "500",
              },
              Object {
                "color": "rgba(0, 0, 0, 1)",
              },
            ],
          ]
        }
      >
        Recovery words
      </Text>
      <Text />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderBottomColor": "rgba(229, 229, 229, 1)",
          "borderBottomWidth": 1,
          "borderLeftColor": "rgba(229, 229, 229, 1)",
          "borderRightColor": "rgba(229, 229, 229, 1)",
          "borderTopColor": "rgba(229, 229, 229, 1)",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 8,
          "paddingTop": 16,
        }
      }
      testID="passcode_faq"
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "fontFamily": "MediumFont",
                "fontWeight": "500",
              },
              Object {
                "color": "rgba(0, 0, 0, 1)",
              },
            ],
          ]
        }
      >
        Passcode
      </Text>
      <Text />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderBottomColor": "rgba(229, 229, 229, 1)",
          "borderBottomWidth": 1,
          "borderLeftColor": "rgba(229, 229, 229, 1)",
          "borderRightColor": "rgba(229, 229, 229, 1)",
          "borderTopColor": "rgba(229, 229, 229, 1)",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 8,
          "paddingTop": 16,
        }
      }
      testID="utxo_and_token_faq"
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "fontFamily": "MediumFont",
                "fontWeight": "500",
              },
              Object {
                "color": "rgba(0, 0, 0, 1)",
              },
            ],
          ]
        }
      >
        UTXO vs Token
      </Text>
      <Text />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderBottomColor": "rgba(229, 229, 229, 1)",
          "borderBottomWidth": 1,
          "borderLeftColor": "rgba(229, 229, 229, 1)",
          "borderRightColor": "rgba(229, 229, 229, 1)",
          "borderTopColor": "rgba(229, 229, 229, 1)",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 8,
          "paddingTop": 16,
        }
      }
      testID="dex_faq"
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "fontFamily": "MediumFont",
                "fontWeight": "500",
              },
              Object {
                "color": "rgba(0, 0, 0, 1)",
              },
            ],
          ]
        }
      >
        DEX
      </Text>
      <Text />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderBottomColor": "rgba(229, 229, 229, 1)",
          "borderBottomWidth": 1,
          "borderLeftColor": "rgba(229, 229, 229, 1)",
          "borderRightColor": "rgba(229, 229, 229, 1)",
          "borderTopColor": "rgba(229, 229, 229, 1)",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 8,
          "paddingTop": 16,
        }
      }
      testID="liquidity_mining_faq"
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "fontFamily": "MediumFont",
                "fontWeight": "500",
              },
              Object {
                "color": "rgba(0, 0, 0, 1)",
              },
            ],
          ]
        }
      >
        Liquidity Mining
      </Text>
      <Text />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderBottomColor": "rgba(229, 229, 229, 1)",
          "borderBottomWidth": 1,
          "borderLeftColor": "rgba(229, 229, 229, 1)",
          "borderRightColor": "rgba(229, 229, 229, 1)",
          "borderTopColor": "rgba(229, 229, 229, 1)",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 8,
          "paddingTop": 16,
        }
      }
      testID="loans_faq"
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "fontFamily": "MediumFont",
                "fontWeight": "500",
              },
              Object {
                "color": "rgba(0, 0, 0, 1)",
              },
            ],
          ]
        }
      >
        Loans
      </Text>
      <Text />
    </View>
  </View>
</RCTScrollView>
`;
